pochivm/api_base.h

```diff
@@ -117,111 +117,223 @@ Value<T>::operator Value<U>() const
 
 // Arithmetic ops convenience operator overloading
 //
-template<typename T, typename = std::enable_if_t<
-             AstTypeHelper::primitive_type_supports_binary_op<T, AstTypeHelper::BinaryOps::ADD>::value> >
-Value<T> operator+(const Value<T>& lhs, const Value<T>& rhs)
+template <AstTypeHelper::BinaryOps Binop, typename T, typename U,
```

Comment:
What is the point of this 'Binop' if you are having an 'expr_type' parameter already? This is extremely fragile.